src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/PhillipsBrineDensity.hpp

```diff
@@ -243,37 +244,41 @@ void PhillipsBrineDensityUpdate::compute( real64 const & pressure,
 
   // Brine density
   // equation (1) from Garcia (2001)
-  if( useMass )
-  {
-    value = density
-            + m_componentMolarWeight[m_CO2Index] * conc
-            - concDensVol;
-    dValue[Deriv::dP] = densityDeriv[0]
-                        + m_componentMolarWeight[m_CO2Index] * dConc_dPres
-                        - dConcDensVol_dPres;
-    dValue[Deriv::dT] = densityDeriv[1]
-                        + m_componentMolarWeight[m_CO2Index] * dConc_dTemp
-                        - dConcDensVol_dTemp;
-    dValue[Deriv::dC+m_CO2Index] = m_componentMolarWeight[m_CO2Index] * dConc_dComp[m_CO2Index]
-                                   - dConcDensVol_dComp[m_CO2Index];
-    dValue[Deriv::dC+m_waterIndex] = m_componentMolarWeight[m_CO2Index] * dConc_dComp[m_waterIndex]
-                                     - dConcDensVol_dComp[m_waterIndex];
-  }
-  else
+  value = density
+          + m_componentMolarWeight[m_CO2Index] * conc
+          - concDensVol;
+  dValue[Deriv::dP] = densityDeriv[0]
+                      + m_componentMolarWeight[m_CO2Index] * dConc_dPres
+                      - dConcDensVol_dPres;
+  dValue[Deriv::dT] = densityDeriv[1]
+                      + m_componentMolarWeight[m_CO2Index] * dConc_dTemp
+                      - dConcDensVol_dTemp;
+  dValue[Deriv::dC+m_CO2Index] = m_componentMolarWeight[m_CO2Index] * dConc_dComp[m_CO2Index]
+                                 - dConcDensVol_dComp[m_CO2Index];
+  dValue[Deriv::dC+m_waterIndex] = m_componentMolarWeight[m_CO2Index] * dConc_dComp[m_waterIndex]
+                                   - dConcDensVol_dComp[m_waterIndex];
+  if( !useMass )
   {
-    value = density / m_componentMolarWeight[m_waterIndex]
-            + conc
-            - concDensVol / m_componentMolarWeight[m_waterIndex];
-    dValue[Deriv::dP] = densityDeriv[0]  / m_componentMolarWeight[m_waterIndex]
-                        + dConc_dPres
-                        - dConcDensVol_dPres / m_componentMolarWeight[m_waterIndex];
-    dValue[Deriv::dT] = densityDeriv[1]  / m_componentMolarWeight[m_waterIndex]
-                        + dConc_dTemp
-                        - dConcDensVol_dTemp  / m_componentMolarWeight[m_waterIndex];
-    dValue[Deriv::dC+m_CO2Index] = dConc_dComp[m_CO2Index]
-                                   - dConcDensVol_dComp[m_CO2Index] / m_componentMolarWeight[m_waterIndex];
-    dValue[Deriv::dC+m_waterIndex] = dConc_dComp[m_waterIndex]
-                                     - dConcDensVol_dComp[m_waterIndex] / m_componentMolarWeight[m_waterIndex];
+    real64 const MT =
```

Comment:
@klevzoff @joshua-white @dkachuma @jafranc also here

Reply:
Looks right to me

Reply:
Can we think of a refactor (maybe left for later) in which the duplicated blocks are factored out in one place ? I start finding difficult to maintain that as little details will eventually diverged

Reply:
moved common parts into PVTFunctionBaseUpdate